src/components/DailyConfigComponent.tsx

```diff
@@ -51,91 +72,171 @@ const DailyConfig = () => {
-      }}
-    >
-      <Switch
-        defaultChecked={config.enabled}
-        onChange={(e) => {
-          const enabled = (e.target as HTMLInputElement).checked;
-          if (enabled) {
-            saveDailyConfig({
-              "workflow name": workflowName || "Daily",
-            });
-            scheduleNextDailyRun({ tomorrow: true });
-          } else {
-            window.clearTimeout(getDailyConfig()["next-run-timeout"]);
-            saveDailyConfig({ "workflow name": "" });
-            setWorkflowName("");
-          }
-          setDisabled(!enabled);
-        }}
-        label={disabled ? "Disabled" : "Enabled"}
-      />
-      <Label>
-        Workflow Name
-        <InputGroup
-          value={workflowName}
-          onChange={(e) => {
-            saveDailyConfig({
-              "workflow name": e.target.value,
-            });
-            setWorkflowName(e.target.value);
+    <div>
+      <Tabs className="roamjs-daily-config-tabs" defaultSelectedTabId="dct">
+        {enabled && (
+          <Tab
+            className="text-white"
+            id="dct"
+            title="Daily Config"
+            panel={
+              <div className="flex items-start gap-2 flex-col">
+                <FormGroup
+                  label="Workflow Name"
+                  labelFor="roamjs-workflow-name"
+                  className="my-4"
+                >
+                  <InputGroup
+                    id="roamjs-workflow-name"
+                    value={workflowName}
+                    onChange={(e) => {
+                      setConfig({ "workflow name": e.target.value });
+                    }}
+                    style={{ minWidth: "initial" }}
+                    placeholder={"Enter workflow name"}
+                  />
+                </FormGroup>
+
+                <FormGroup
+                  label="Time To Run"
+                  labelFor="roamjs-time-picker"
+                  className="mb-4 flex"
+                  style={{ alignItems: "center" }}
+                  inline={true}
+                >
+                  <div className="flex items-center">
+                    <TimePicker
+                      value={timePicker}
+                      onChange={(date) => setTimePicker(date)}
+                      showArrowButtons
+                      disabled={!isEditTime}
+                      className={"user-select-none flex-1 text-center"}
+                    />
+                    <Icon
+                      icon={isEditTime ? "tick" : "edit"}
+                      onClick={async () => {
+                        if (isEditTime) {
+                          const newTime = `${timePicker.getHours()}:${timePicker.getMinutes()}`;
+                          await setConfig({ time: newTime });
+                          await scheduleNextDailyRun({ tomorrow: true });
+                          _setLocalConfig(getDailyConfig()); // force update, scheduleNextDailyRun runs saveDailyConfig
```

Comment:
This doesn't seem ideal.

Reply:
Running into some type errors up the chain

adding `Promise<DailyConfig>` to `scheduleNextDailyRun` and `getDailyConfig` still left `getExtensionApi().settings.set` returning `void`, so left it as is